src/Controller/ProductsController.php

```diff
@@ -38,18 +40,15 @@ public function __construct(private SerializerInterface $serializer, private Pro
      * @OA\Tag(name="Products")
      */
     #[Route('/api/products', methods: ['GET'], name: 'products_show')]
-    public function showProducts(Request $request): JsonResponse
+    public function showProducts(Request $request): Response
     {
-        $productsList = $this->productManager->getProductsList();
+        $products = $this->productManager->getProductsList();
 
-        $productsList = $this->paginatorApi->paginate(
-            $request,
-            $productsList
-        );
+        $productsList = $this->paginator->paginate($products, $request->query->getInt('page', 1), 10);
 
-        $this->serializer->serialize($productsList, 'json', ['groups' => 'show_products']);
+        $json = $this->serializer->serialize($productsList, 'json');
```

Comment:
As tu essayé la gestion des groupes de la librairie ? Il faut voir dans la documentation officielle de JMS Serializer : 
- https://jmsyst.com/libs/serializer/master/cookbook/exclusion_strategies
- https://jmsyst.com/libs/serializer/master/reference/annotations

```suggestion
        $json = $this->serializer->serialize($productsList, 'json', SerializationContext::create()->setGroups(['show_products']));
```

Reply:
done